azurecontainerapps/azurecontainerapps.ts

```diff
@@ -1,18 +1,31 @@
+import * as fs from 'fs';
 import * as path from 'path';
 import * as tl from 'azure-pipelines-task-lib/task';
-import * as fs from 'fs';
-import { Utility } from './src/Utility';
-import { ContainerAppHelper } from './src/ContainerAppHelper';
 import { AzureAuthenticationHelper } from './src/AzureAuthenticationHelper';
+import { ContainerAppHelper } from './src/ContainerAppHelper';
 import { ContainerRegistryHelper } from './src/ContainerRegistryHelper';
+import { TelemetryHelper } from './src/TelemetryHelper';
+import { Utility } from './src/Utility';
 
 const util = new Utility();
 
 export class azurecontainerapps {
 
     public static async runMain(): Promise<void> {
+        let disableTelemetry = false;
+        try {
+            disableTelemetry = tl.getBoolInput('disableTelemetry', false);
+        }
+        finally {}
+
+        // Set up TelemetryHelper for managing telemetry calls
+        const telemetryHelper: TelemetryHelper = new TelemetryHelper(disableTelemetry);
```

Comment:
Should we almost immediately have an "Initialize" call logged as telemetry, that would allow us to detect any issues with the task? For example, if we see a lot more Initialize calls than the sum of the other calls, we know we're losing something, and we likely have some errors in the task. What do you think?

Reply:
Oh, I see we're building the log and then logging once at the end. I'm wondering if we should at least log 2 events (1 Initialize, 1 as it is today) so that we can validate that what we have is reliable, and not just showing the successful calls.

Reply:
In the `TelemetryHelper` class, the `result` is "failed" by default and only gets set to "succeeded" when the `setSuccessfulResult()` method is called after `az containerapp up` is called -- this means that if anything fails _before_ (and during) the call to `az cxontainerapp up`, we're going to end up logging that the result was "failed", which should give us the same insight as if we had the two calls that you're proposing (minus the additional cost of calling `oryx telemetry` an additional time and managing an additional event) -- what do you think?

Reply:
Ah yes, the fact that we docker run the image would make this more costly than it should for a second call. Looking at the code, we're wrapping almost all the code in the big try catch finally block, so that's likely OK.